airbyte-integrations/connectors/source-gainsight-cs/source_gainsight_cs/streams.py

```diff
@@ -0,0 +1,116 @@
+from abc import ABC
+from typing import Any, Iterable, Mapping, MutableMapping, Optional
+
+import requests
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+class GainsightCsStream(HttpStream, ABC):
+    def __init__(self, domain_url: str, **kwargs):
+        super().__init__(**kwargs)
+        self.domain_url = f"{domain_url}/v1/"
+
+    @property
+    def url_base(self):
+        return self.domain_url
+
+
+class GainsightCsObjectStream(GainsightCsStream):
+    primary_key = "gsid"
+    limit = 100
+    json_schema = None
+    offset = 0
+
+    gainsight_airbyte_type_map = {
+        "STRING": ["null", "string"],
+        "BOOLEAN": ["null", "boolean"],
+        "NUMBER": ["null", "number"],
+        "PERCENTAGE": ["null", "number"],
+        "CURRENCY": ["null", "number"],
+        "GSID": ["null", "string"],
+        "DATETIME": ["null", "string"],
+        "EMAIL": ["null", "string"],
+        "URL": ["null", "string"],
+        "RICHTEXTAREA": ["null", "string"],
+        "LOOKUP": ["null", "string"],
+        "JSON": ["null", "object"],
+        "JSONBOOLEAN": ["null", "boolean"],
+        "JSONNUMBER": ["null", "number"],
+        "JSONSTRING": ["null", "string"]
+    }
+
+    def __init__(self, name: str, domain_url: str, **kwargs):
+        super().__init__(domain_url, **kwargs)
+        self.object_name = name
+
+    @property
+    def name(self):
+        return self.object_name
+
+    def lowercase(self, field_name):
+        return field_name[0].lower() + field_name[1:]
+
+    def dynamic_schema(self, full_schema, metadata):
+        for field in metadata:
+            field_name = self.lowercase(field['fieldName'])
```

Comment:
Honestly don't remember. I think it was to try to be standardized because some fields were lowercased and some uppercased but I don't think it is needed so I removed it.